plugins/notification-mappers/src/index.ts

```diff
@@ -0,0 +1,190 @@
+    objectId: ctx.objectId,
+    objectClass: ctx.objectClass,
+    _id: `dd-spaceInfo-${ctx.objectId}` as Ref<SpaceInfo>,
+    modifiedBy: ctx.modifiedBy,
+    modifiedOn: ctx.modifiedOn,
+    createOn: ctx.createOn,
+    space: ctx.objectId,
+    descriptorId: descriptorId,
+    lastModified: ctx.modifiedOn
+  }
+  return data
+}
+
+export default (): void => {
+  registerMapper(notification.mappers.SpaceInfo, {
+    map: async (tx: Tx, options: MappingOptions): Promise<SpaceInfo[]> => {
+      if (tx._class === core.class.TxCreateDoc) {
+        const ctx = tx as TxCreateDoc<Doc>
+        if (options.hierarchy.isDerived(ctx.objectClass, core.class.Space)) {
+          return [createSpaceInfo(ctx as TxCreateDoc<Space>, options.descriptor._id)]
+        } else if (
+          ctx.objectSpace !== core.space.Model &&
+          !options.hierarchy.isDerived(ctx.objectClass, core.class.DerivedData)
+        ) {
+          await updateSpaceInfo(ctx, options)
+        }
+      }
+
+      if (tx._class === core.class.TxUpdateDoc) {
+        const ctx = tx as TxUpdateDoc<Doc>
+        if (options.hierarchy.isDerived(ctx.objectClass, core.class.Space)) {
+          const result = await options.storage.findAll(notification.class.SpaceInfo, { objectId: ctx.objectId })
+          if (result.length === 0) {
+            return [createSpaceInfo(ctx as TxUpdateDoc<Space>, options.descriptor._id)]
+          }
+          return result
+        } else if (
+          ctx.objectSpace !== core.space.Model &&
+          !options.hierarchy.isDerived(ctx.objectClass, core.class.DerivedData)
+        ) {
+          await updateSpaceInfo(ctx, options)
+        }
+      }
+
+      if (tx._class === core.class.TxRemoveDoc) {
+        const ctx = tx as TxRemoveDoc<Doc>
+        if (
+          !options.hierarchy.isDerived(ctx.objectClass, core.class.Space) &&
+          ctx.objectSpace !== core.space.Model &&
+          !options.hierarchy.isDerived(ctx.objectClass, core.class.DerivedData)
+        ) {
+          await updateSpaceInfo(ctx, options)
+        }
+      }
+
+      return []
+    }
+  })
+
+  registerMapper(notification.mappers.SpaceNotification, {
+    map: async (tx: Tx, options: MappingOptions): Promise<SpaceNotifications[]> => {
+      if (tx._class === core.class.TxCreateDoc) {
+        const result: SpaceNotifications[] = []
+        const ctx = tx as TxCreateDoc<Space>
+        const doc = TxProcessor.createDoc2Doc(ctx) as Space
+        for (const member of doc.members) {
+          result.push(createSpaceNotification(ctx, options, member))
+        }
+        return result
+      }
+
+      if (tx._class === core.class.TxUpdateDoc) {
+        const ctx = tx as TxUpdateDoc<Space>
+        if (ctx.operations.members !== undefined) {
+          const result: SpaceNotifications[] = []
+          for (const member of ctx.operations.members) {
+            result.push(createSpaceNotification(ctx, options, member))
+          }
+          return result
```

Comment:
You will loose all existing person notifications on new person join, since DD processor will update all fields to new values. You need to find SpaceNotifications for old members and create only for new ones.

Reply:
Am I already doing it, or not?